### DISCUSSION TITLE
Access dynamic page route in custom server

### DISCUSSION THREAD

**Discussion by brianabaker** [vercel/next.js#29087] [Help]
Hello, is it possible to see what the page route is within a custom server? I'd like to have it for metrics logging. Thanks! 

Edit: Specifically looking for the dynamic route e.g. `posts/[id]` when in production.

**Comment by wilsonneto-swe** [comment]
Looking at the below example I think can helop you:
```
// server.js
const { createServer } = require('http')
const { parse } = require('url')
const next = require('next')

const dev = process.env.NODE_ENV !== 'production'
const app = next({ dev })
const handle = app.getRequestHandler()

app.prepare().then(() => {
  createServer((req, res) => {
    // Be sure to pass `true` as the second argument to `url.parse`.
    // This tells it to parse the query portion of the URL.
    const parsedUrl = parse(req.url, true)
    const { pathname, query } = parsedUrl

    if (pathname === '/a') {
      app.render(req, res, '/a', query)
    } else if (pathname === '/b') {
      app.render(req, res, '/b', query)
    } else {
      handle(req, res, parsedUrl)
    }
  }).listen(3000, (err) => {
    if (err) throw err
    console.log('> Ready on http://localhost:3000')
  })
})
```
The pathname variable has the page. Is this what you need?

I hope it helps you.
Thank you very much, success on your project!
Beste regards

**Reply by brianabaker** [reply]
Thanks! I should have specified I'm using dynamic routing e.g. `posts/[id]`. I'm looking to find the dynamic route itself. 

Locally, I can find the dynamic route through `query: { page: posts/[id] }`, but in production I don't know how to find that information, it's not there it seems, unless I'm missing something! `pathname` doesn't suffice due to the dynamic route I'm looking for.
